### PR TITLE
Implement sbt build module extension point

### PR DESCRIPTION
**Pull request by AnthonyGrod**

### CODE REVIEW

File: scala/integration/intellij-bsp/src/org/jetbrains/plugins/scala/project/bsp/sbt/module/SbtBspModuleSettingsEditor.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        val sbtVersion = sbtBuildTarget.getSbtVersion
+        val imports = sbtBuildTarget.getAutoImports.asScala.flatMap(_.split(", ")).asJava
+        val resolversModel = new ResolversModel(resolvers, state.getProject)
+
+        myForm.reset(sbtVersion, imports, resolversModel)
+    }
+  }
+}
+
+private class ResolversModel(val resolvers: Seq[SbtResolver], val project:Project) extends AbstractTableModel {
```
Comment (agluszak) [inline scala/integration/intellij-bsp/src/org/jetbrains/plugins/scala/project/bsp/sbt/module/SbtBspModuleSettingsEditor.scala:42]:
```suggestion
private class ResolversModel(val resolvers: Seq[SbtResolver], val project: Project) extends AbstractTableModel {
```
please run scalafmt

File: scala/integration/intellij-bsp/src/org/jetbrains/plugins/scala/project/bsp/sbt/module/SbtBspModuleSettingsEditor.scala [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  override def getColumnName(columnIndex: Int): String = columns(columnIndex)
+
+  override def getValueAt(rowIndex: Int, columnIndex: Int): String = {
+    val valueOpt = columnIndex match {
+      case 0 => resolvers.lift(rowIndex).map(_.name)
+      case 1 => resolvers.lift(rowIndex).map(_.root)
+      case _ => Some("")
+    }
+    valueOpt.getOrElse("???")
```
Comment (agluszak) [inline scala/integration/intellij-bsp/src/org/jetbrains/plugins/scala/project/bsp/sbt/module/SbtBspModuleSettingsEditor.scala:61]: why use option then?

Comment (AnthonyGrod) [reply scala/integration/intellij-bsp/src/org/jetbrains/plugins/scala/project/bsp/sbt/module/SbtBspModuleSettingsEditor.scala:61]: I saw that the settings editor logic in sbt module was written in the same manner. As for my understanding, there should be a difference between going out of the column bounds and the row bounds. The difference is that in the first case an empty string is returned and in the second case "???" is returned.

**Comment by AnthonyGrod** [comment]
Copy of the PR with updated base here and performed changes here: https://github.com/mapz-zpp/intellij-scala/pull/29

**Pull request closed without merging**